### PR TITLE
Ns filter wizard content

### PR DESCRIPTION
**Pull request by natibekele**
filter languages on project creation for only languages loaded into database

---
This change is [Reviewable](https://reviewable.io/reviews/wycliffeassociates/otter-jvm/450)

### CODE REVIEW

File: src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt [hunk 1]
```diff
@@ -52,6 +55,17 @@ class ProjectWizardViewModel : ViewModel() {
...
     }
 
+    private fun filterToAvailableLanguges() {
+        collectionRepo
+            .getRootSources()
+            .observeOnFx()
+            .subscribe { retrieved ->
+                retrieved.forEach {collection ->
+                    filteredLanguages.addAll(languages.filter { it.id == collection.resourceContainer?.language?.id })
+                }
```
Comment (aunger) [inline src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt]: This looks like it'll create duplicate entries in filteredLanguages if you have multiple resource containers in the same language. I suggest you make a set of languages from the RCs, and use that set to filter languages.

File: src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt [hunk 2]
```diff
@@ -52,6 +55,17 @@ class ProjectWizardViewModel : ViewModel() {
         }
     }
 
+    private fun filterToAvailableLanguges() {
```
Comment (aunger) [inline src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt]: Although `filterToAvailableLanguges` is misspelled, it's nice and descriptive.  Consider renaming the `filteredLanguages` field to `availableLanguages` so they match.

**Review by aunger — CHANGES_REQUESTED** [review]

**Review by natibekele — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/wycliffeassociates/otter-jvm/450)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @aunger)

___
*[src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt, line 58 at r1](https://reviewable.io/reviews/wycliffeassociates/otter-jvm/450#-LmkBUJeba4kkA8xylSW-r1-58:-LmkbwTo9OVXURLGpnPB:b-896fix) ([raw file](https://github.com/wycliffeassociates/otter-jvm/blob/78d811ca72649ab834b130c77724e03da1199c49/src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt#L58)):*
<details><summary><i>Previously, aunger (Russell Aunger) wrote…</i></summary><blockquote>

Although `filterToAvailableLanguges` is misspelled, it's nice and descriptive.  Consider renaming the `filteredLanguages` field to `availableLanguages` so they match.
</blockquote></details>

Done.
___
*[src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt, line 65 at r1](https://reviewable.io/reviews/wycliffeassociates/otter-jvm/450#-LmkBUJeba4kkA8xylSW-r1-65:-LmkbysV9_2hqiB5hZZH:b-896fix) ([raw file](https://github.com/wycliffeassociates/otter-jvm/blob/78d811ca72649ab834b130c77724e03da1199c49/src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt#L65)):*
<details><summary><i>Previously, aunger (Russell Aunger) wrote…</i></summary><blockquote>

This looks like it'll create duplicate entries in filteredLanguages if you have multiple resource containers in the same language. I suggest you make a set of languages from the RCs, and use that set to filter languages.
</blockquote></details>

Done.

**Review by aunger — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/wycliffeassociates/otter-jvm/450)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @aunger and @natibekele)

___
*[src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt, line 68 at r4](https://reviewable.io/reviews/wycliffeassociates/otter-jvm/450#-LmllkcT0WviPfiybVHg:-LmllkcU4lmY4-frPnZt:bio4al5) ([raw file](https://github.com/wycliffeassociates/otter-jvm/blob/ea8d7fabf664078fc2bc6d22ad898b83b05e8129/src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt#L68)):*
> ```Kotlin
>             }.map { uniqueLanguages ->
>                 filteredLanguages.setAll(uniqueLanguages)
>             }.subscribe()
> ```

Side effects should only be called from `.subscribe` or one of the `.do*` methods on Observable (never `.map`).  In this case, just combine the last `map` and `subscribe` into a `subscribe`.  More reading if you care: [1](https://www.grokkingandroid.com/rxjavas-side-effect-methods/), [2](http://introtorx.com/Content/v1.0.10621.0/09_SideEffects.html)

**Review by natibekele — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/wycliffeassociates/otter-jvm/450)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @aunger)

___
*[src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt, line 68 at r4](https://reviewable.io/reviews/wycliffeassociates/otter-jvm/450#-LmllkcT0WviPfiybVHg:-LmoAuqh0ddA7k7eOTv-:b7w8ku5) ([raw file](https://github.com/wycliffeassociates/otter-jvm/blob/ea8d7fabf664078fc2bc6d22ad898b83b05e8129/src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt#L68)):*
<details><summary><i>Previously, aunger (Russell Aunger) wrote…</i></summary><blockquote>

Side effects should only be called from `.subscribe` or one of the `.do*` methods on Observable (never `.map`).  In this case, just combine the last `map` and `subscribe` into a `subscribe`.  More reading if you care: [1](https://www.grokkingandroid.com/rxjavas-side-effect-methods/), [2](http://introtorx.com/Content/v1.0.10621.0/09_SideEffects.html)
</blockquote></details>

this appears to be an older commit. I have already refactored that function and chained all the additional work together

**Review by natibekele — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/wycliffeassociates/otter-jvm/450)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @aunger)

___
*[src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt, line 68 at r4](https://reviewable.io/reviews/wycliffeassociates/otter-jvm/450#-LmllkcT0WviPfiybVHg:-LmoBqoL4SumOxeynJgA:bg7hxh2) ([raw file](https://github.com/wycliffeassociates/otter-jvm/blob/ea8d7fabf664078fc2bc6d22ad898b83b05e8129/src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/projectwizard/viewmodel/ProjectWizardViewModel.kt#L68)):*
<details><summary><i>Previously, natibekele wrote…</i></summary><blockquote>

this appears to be an older commit. I have already refactored that function and chained all the additional work together
</blockquote></details>

nevermind. the comment was showing up with an older commit in reviewable. i'll take a look at it now

**Review by aunger — APPROVED** [review]
> Reviewed 1 of 2 files at r3, 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/wycliffeassociates/otter-jvm/450)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**